tests/approx_matching_ut.cpp

```diff
@@ -200,4 +200,52 @@ SIMPLE_UNIT_TEST_SUITE(ApproxMatchingTest) {
 			ACCEPTS("");
 		}
 	}
+
+	SIMPLE_UNIT_TEST(StressTest) {
+		ystring regexp = "^";
+		for (size_t letter = 0; letter < 10; ++letter) {
+			regexp += ystring(3, letter + 'a');
+		}
+		regexp += "$";
+
+		auto fsm = BuildFsm(regexp.data());
+
+		ystring changedRegexp = regexp.substr(1, regexp.size() - 2);
+
+		APPROXIMATE_SCANNER(fsm, 1) {
+			for (size_t pos = 0; pos < regexp.size() - 2; ++pos) {
+				changedRegexp[pos] = 'x';
+				ACCEPTS(changedRegexp);
+
+				changedRegexp.erase(pos, 1);
+				ACCEPTS(changedRegexp);
+				changedRegexp.insert(pos, 1, regexp[pos + 1]);
+
+				changedRegexp.insert(pos, 1, 'x');
+				ACCEPTS(changedRegexp);
+				changedRegexp.erase(pos, 1);
+			}
+		}
+
+		APPROXIMATE_SCANNER(fsm, 2) {
```

Comment:
IMHO
It is worth to check that the same mutations are denied for the `APPROXIMATE_SCANNER(fsm, 1)`.